zlib-rs/src/deflate/trees_tbl.rs

```diff
@@ -70,6 +70,27 @@ pub const STATIC_LTREE: [Value; L_CODES + 2] = [
     h(163,8), h( 99,8), h(227,8)
 ];
 
+/// Precomputes the `Values` generated by the `encode_len` function for all `u8` input values
+#[rustfmt::skip]
+pub const STATIC_LTREE_ENCODINGS: [Value; 256] = {
+    let mut table = [Value::new(0, 0); 256];
+
+    let mut lc = 0;
+    while lc < table.len() {
+        let (code, len) = super::encode_len(&STATIC_LTREE, lc as u8);
+
+        // assert that there is no precision loss
+        assert!(code as u16 as u64 == code);
+        assert!(len as u16 as usize == len);
```

Comment:
The following would have been nicer, but unfortunately rust doesn't yet support calling trait methods in const contexts.

```suggestion
        assert!(u16::try_from(code).is_ok());
        assert!(u16::try_from(len).is_ok());
```

Reply:
unfortunately this is a `const fn`,and so trait methods don't work.

Reply:
Indeed